### ISSUE TITLE
Update to html

### ISSUE THREAD

**Issue by fdncred** [nushell/wasm-nu#5]
Can you bring over the new `to html` code that was committed yesterday? I'd like to see what the new `to html --html_color` and `to html --no_color` does for things like this.

**Comment by jzaefferer** [comment]
We currently build nu from a special branch:
```
nu-cli = {git = "https://github.com/jonathandturner/nushell.git", branch = "wasm_wip", default-features = false}
```
I think @jonathandturner would have to update that, or merge it to master so that we can build wasm-nu from there.

**Comment by jonathandturner** [comment]
I've updated it. I filed a bug on the nushell repo that's stopping us from moving now:

https://github.com/nushell/nushell/issues/2187

**Comment by fdncred** [comment]
For a quick work-around I changed:
* index.html line to this `<link rel="stylesheet" href="https://newcss.net/theme/boilerplate.css" />`
* index.js line to this ` return run_nu(input + "| to html --html_color");`
Not perfect but better than black text on a black background :)
![image](https://user-images.githubusercontent.com/343840/87566015-06ca6200-c688-11ea-8301-efd683e13c62.png)

**Comment by jzaefferer** [comment]
@fdncred I've applied those changes in 83f8243cc0f2da93e0308fbd2788930f6de29025 and a113f16c1080b53e495656f30795e88d83c71368

As your screenshot already shows, some of those colors have very poor contrast (the light blue; the yellow is worst). How/where can we improve those?

**Comment by fdncred** [comment]
@jzaefferer Agreed. This can be partially addressed with the new PR I just landed in nushell which adds the `--dark_bg` to `to html`. `--dark_bg` means you are telling `nu` that you have a dark background and so it should invert the black and white colors. If you don't specifiy `--dark_bg` then the colors look like above because it assumes you have a light background.

Changing the colors of yellow is not really in the purview of `to html` at this time because it infers there are themes. For instance, this is why Windows Terminal supports [color schemes](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/color-schemes). If we had color theming, then we could change the yellow to more of a mustard color, or something like that.

It's yellow because anis escape sequence 33 calls for yellow. The ansi escape sequence doesn't account for backgrounds.

Having said all that, it's a simple change in the `to html` code to change all the places we use `color:yellow` to `color:you_pick`. We could even do that only if the background color is not dark. I'm happy to make that change if @jonathandturner wants it. Someone just needs to specify what color yellow should be on a light background.

**Comment by jzaefferer** [comment]
Testing alternatives with the Firefox DevTools, which check contrast according to WCAG, we could replace `color: cyan;` with `color: #037979` and `color: yellow` with `color: #717100`.

The magenta didn't look so bad, but should also be a bit darker to get within the WCAG AA range, to `#c800c8`.

I'd be happy to test the `--dark_bg` option once available, too.

**Comment by fdncred** [comment]
@jzaefferer Sounds good to me. @jonathandturner Thoughts on changing the colors in `to html` as per Jörn's suggestion?

**Comment by jonathandturner** [comment]
@jonathandturner sounds good

**Comment by fdncred** [comment]
Implemented as stated above https://github.com/nushell/nushell/pull/2200. @jzaefferer Once this PR lands and Jonathan's wasm PR lands, yes, I think it would be cool to test the `--dark_bg` to see what happens.  BTW - that yellow you chose looks like the color of baby poop. LOL. Just saying.

**Comment by jzaefferer** [comment]
I tried `--dark_bg` with the terminal theme:

<img width="675" alt="Screenshot 2020-07-18 at 10 31 27" src="https://user-images.githubusercontent.com/52585/87847550-db42b400-c8e1-11ea-9fb6-e11b818036cc.png">

Not bad, but looks like most of the colors have too little contrast again. As far as I understand it, we'd have to extend `fn setup_html_color_regexes` to use different colors for the individual hash items, depending on `is_dark`.

If that makes sense, I can try to suggest updated colors for this dark mode, too. Unless we have someone in the community who can do that better than me (pontaoski blackquill?)

**Comment by fdncred** [comment]
@jzaefferer I already have an `if is_dark` section, so I can make those the original colors which look good on black. Preferably this output would look identical to the regular terminal output which is what this screenshot is from.
![image](https://user-images.githubusercontent.com/343840/87852694-cfe38e80-c8c9-11ea-885e-085e3d3b4951.png)

**Comment by fdncred** [comment]
I landed https://github.com/nushell/nushell/pull/2212 that hopefully helps this.

**Comment by jzaefferer** [comment]
I think this is addressed. There's still discussion going on in https://github.com/nushell/nushell/pull/2209 - whatever the result of that will make it back to this repo anyway.
